plugins/inputs/windows_event_log/wineventlog/utils.go

```diff
@@ -112,8 +105,31 @@ func UTF16ToUTF8Bytes(in []byte, length uint32) ([]byte, error) {
 	return decoded, err
 }
 
+func UTF16ToUTF8BytesForWindowsEventBuffer(in []byte, length uint32) ([]byte, error) {
+	// Since Windows server 2022, the returned value of used buffer represents for double bytes char count,
+	// which is half of the actual buffer used by byte(what older Windows OS returns), checking if the length
+	//land on the end of used buffer, if no, double it.
+	if NumbersOfBytesPerCharacter == UnknownBytesPerCharacter {
+		if isTheEndOfContent(in, length) {
+			log.Printf("I! Buffer used: %d is returning as single byte character count", length)
+			NumbersOfBytesPerCharacter = 1
+		} else {
+			log.Printf("I! Buffer used: %d is returning as double byte character count, doubling it to get the whole buffer content.", length)
+			NumbersOfBytesPerCharacter = 2
```

Comment:
Thinking about this more now, if we double the number of characters that get read in, does this mean that compared to Windows 2019, CWA running on Windows 2022 would potentially use twice as much memory when consuming Windows event logs?

Reply:
No, they consumed same amount of space, the API behavior change is Windows 2022 decides to return n for a message with n characters while Windows 2019 returns 2n, because they are 2-bytes characters.